code/src/nuvla/job/actions/nuvlabox_reboot.py

```diff
@@ -1,28 +1,43 @@
 # -*- coding: utf-8 -*-
-
+import os
 import logging
 
 from ..actions import action
-from ...connector import nuvlabox as NB
+from ...connector.nuvlabox import NuvlaBox
+from ...connector.kubernetes import K8sEdgeMgmt
+from ...job.job import Job
 
 
 @action('reboot_nuvlabox', True)
 class NBRebootJob(object):
 
-    def __init__(self, _, job):
+    def __init__(self, _, job: Job):
         self.job = job
         self.api = job.api
 
     def reboot(self):
-        nuvlabox_id = self.job['target-resource']['href']
 
-        logging.info('Rebooting NuvlaBox {}.'.format(nuvlabox_id))
-        connector = NB.NuvlaBox(api=self.api, nuvlabox_id=nuvlabox_id, job=self.job)
+        # logging.debug('Rebooting NuvlaEdge %s', self.job.nuvlaedge_id)
+
+        if os.getenv('KUBERNETES_SERVICE_HOST'):
+            ret = self._reboot_k8s()
+        else:
+            ret = self._reboot_docker()
 
-        # IMPORTANT BIT THAT MUST CHANGE FOR EVERY NUVLABOX API ACTION
-        r = connector.reboot()
+        logging.info(f'Reboot action called. Output: {ret}')
 
         return 0
 
+    def _reboot_docker(self) -> str:
+        connector = NuvlaBox(api=self.api, job=self.job)
```

Comment:
@giovannibianco you still need to add back the `nuvlabox_id` query parameter as it was on the previous line 20.
So something like:
```python
NuvlaBox(api=self.api, 
         nuvlabox_id=self.job['target-resource']['href'], 
         job=self.job)